### ISSUE TITLE
Windows Grains doesn't include default gateway

### ISSUE THREAD

**Issue by doesitblend** [saltstack/salt#51128]
### Description of Issue/Question
A comparison between Linux and Windows grains in 2018.3.3
### Setup
Install Salt 2018.3.3 on Windows 2016 server

### Steps to Reproduce Issue
```
[root@winmaster vagrant]# salt '*' grains.item ip4_gw ip_gw ip6_gw
winmaster: # A RHEL Based host and my master 
    ----------
    ip4_gw:
        10.0.2.2
    ip6_gw:
        False
    ip_gw:
        True
win2016-2:
    ----------
    ip4_gw:
    ip6_gw:
    ip_gw:
win2016-3:
    ----------
    ip4_gw:
    ip6_gw:
    ip_gw:
```

### Versions Report

```
win2016-2:
    Salt Version:
               Salt: 2018.3.3
     
    Dependency Versions:
               cffi: 1.10.0
           cherrypy: 10.2.1
           dateutil: 2.6.1
          docker-py: Not Installed
              gitdb: 2.0.5
          gitpython: 2.1.3
              ioflo: Not Installed
             Jinja2: 2.9.6
            libgit2: Not Installed
            libnacl: 1.6.1
           M2Crypto: Not Installed
               Mako: 1.0.6
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)]
       python-gnupg: 0.4.1
             PyYAML: 3.12
              PyZMQ: 16.0.3
               RAET: Not Installed
              smmap: 2.0.5
            timelib: 0.2.4
            Tornado: 4.5.1
                ZMQ: 4.1.6
     
    System Versions:
               dist:   
             locale: cp1252
            machine: AMD64
            release: 2016Server
             system: Windows
            version: 2016Server 10.0.14393  Multiprocessor Free
     
win2016-3:
    Salt Version:
               Salt: 2018.3.3
     
    Dependency Versions:
               cffi: 1.10.0
           cherrypy: 10.2.1
           dateutil: 2.6.1
          docker-py: Not Installed
              gitdb: 2.0.5
          gitpython: 2.1.3
              ioflo: Not Installed
             Jinja2: 2.9.6
            libgit2: Not Installed
            libnacl: 1.6.1
           M2Crypto: Not Installed
               Mako: 1.0.6
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
       python-gnupg: 0.4.1
             PyYAML: 3.12
              PyZMQ: 16.0.3
               RAET: Not Installed
              smmap: 2.0.5
            timelib: 0.2.4
            Tornado: 4.5.1
                ZMQ: 4.1.6
     
    System Versions:
               dist:   
             locale: cp1252
            machine: AMD64
            release: 2016Server
             system: Windows
            version: 2016Server 10.0.14393 SP0 Multiprocessor Free
```

**Comment by doesitblend** [comment]
ZD-3165

**Comment by Ch3LL** [comment]
it looks like it runs this binary: `ip_bin = salt.utils.path.which('ip')`

I dont have a windows VM right on hand can you verify if that is available on windows or not?

Also if you can verify this command:

`out = __salt__['cmd.run']([ip_bin, '-' + ip_version, 'route', 'show'])` where ip_version is 4 or 6

**Comment by rdutch** [comment]
The IP cmd doesn't exist, we would need to use a power shell cmd like this:
Get-NetRoute |
    where {$_.DestinationPrefix -eq '0.0.0.0/0'} |
    select { $_.NextHop }

**Comment by maneamarius** [comment]
I worked around this issue for now, by doing this (which is pretty much the same as Rob)

{% set gateway = salt['cmd.shell']('powershell "Get-NetRoute | where {$_.DestinationPrefix -eq \'0.0.0.0/0\'} | select {$_.NextHop}  | ft -hidetableheaders"') | trim -%}

**Comment by doesitblend** [comment]
In a discussion with @twangboy it looks like the default gateway on Windows is actually already being grabbed per interface in `salt.utils.network.win_interfaces`. Adding this single function to an execution module in a Vagrant environment yields the following results:

```
# Execution Module
import salt
import salt.config
import salt.utils.network
import salt.utils.winapi

import wmi


def netinfo():
    '''
    Return windows network interface info
    '''
    data = salt.utils.network.win_interfaces() 
    return data
```

Results from a run of this function
```
win2016-2: #Salt minion with Python 2
    ----------
    Intel(R) PRO/1000 MT Desktop Adapter:
        ----------
        hwaddr:
            08:00:27:5B:C4:19
        inet:
            |_
              ----------
              address:
                  10.0.2.15
              broadcast:
                  10.0.2.2 #Default gateway
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::64b1:4e41:6bf9:6de0
        up:
            True
    Intel(R) PRO/1000 MT Desktop Adapter #2:
        ----------
        hwaddr:
            08:00:27:D9:64:B3
        inet:
            |_
              ----------
              address:
                  192.168.60.14
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter #2
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::e4c3:71af:416:4745
        up:
            True
win2016-3: #Salt minion with Python 3
    ----------
    Intel(R) PRO/1000 MT Desktop Adapter:
        ----------
        hwaddr:
            08:00:27:5B:C4:19
        inet:
            |_
              ----------
              address:
                  10.0.2.15
              broadcast:
                  10.0.2.2 #default gateway
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::9172:45b2:1df4:e3a8
        up:
            True
    Intel(R) PRO/1000 MT Desktop Adapter #2:
        ----------
        hwaddr:
            08:00:27:2B:39:D0
        inet:
            |_
              ----------
              address:
                  192.168.60.13
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter #2
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::dc66:1ea0:663a:c38a
        up:
            True

```

@maneamarius @rdutch Are either one of you able to test the above execution module to verify it returns the default gateway information? If so, then I believe the fix for this will be pretty simple.

**Comment by damon-atkins** [comment]
how about 
netsstat -r
or 
route print

Powershell is two slow

**Comment by rdutch** [comment]
confirmed on my vm's:
```
[root@master bin]# salt win2016 network.interfaces
win2016:
    ----------
    Intel(R) PRO/1000 MT Desktop Adapter:
        ----------
        hwaddr:
            08:00:27:42:F6:FA
        inet:
            |_
              ----------
              address:
                  192.168.50.60
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::80ce:bc31:6b97:9fed
        up:
            True
    Intel(R) PRO/1000 MT Desktop Adapter #2:
        ----------
        hwaddr:
            08:00:27:D7:47:8C
        inet:
            |_
              ----------
              address:
                  10.0.2.15
              broadcast:
                  10.0.2.2
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter #2
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::2590:c2f0:9740:ad53
        up:
            True
[root@master bin]# salt win2012 network.interfaces
win2012:
    ----------
    Intel(R) PRO/1000 MT Desktop Adapter:
        ----------
        hwaddr:
            08:00:27:57:7D:A2
        inet:
            |_
              ----------
              address:
                  192.168.50.70
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::a925:fa23:21fb:33c6
        up:
            True
    Intel(R) PRO/1000 MT Desktop Adapter #2:
        ----------
        hwaddr:
            08:00:27:44:3E:61
        inet:
            |_
              ----------
              address:
                  10.0.2.15
              broadcast:
                  10.0.2.2
              label:
                  Intel(R) PRO/1000 MT Desktop Adapter #2
              netmask:
                  255.255.255.0
        inet6:
            |_
              ----------
              address:
                  fe80::d5a2:bbdc:b9f9:c38c
        up:
            True
```

**Comment by maneamarius** [comment]
Sorry for late reply, I've been in vacation for last month.
Yes, the fix in the PR referenced in #51372 should be good!
Thanks for helping out with this!
